src/v/pandaproxy/rest/proxy.cc

```diff
@@ -9,52 +9,125 @@
 
 #include "pandaproxy/rest/proxy.h"
 
+#include "config/property.h"
+#include "config/rest_authn_endpoint.h"
+#include "net/unresolved_address.h"
 #include "pandaproxy/api/api-doc/rest.json.h"
 #include "pandaproxy/logger.h"
+#include "pandaproxy/parsing/exceptions.h"
+#include "pandaproxy/parsing/from_chars.h"
 #include "pandaproxy/rest/configuration.h"
 #include "pandaproxy/rest/handlers.h"
+#include "pandaproxy/types.h"
+#include "redpanda/request_auth.h"
+#include "utils/gate_guard.h"
 
 #include <seastar/core/future-util.hh>
 #include <seastar/core/memory.hh>
 #include <seastar/http/api_docs.hh>
 
+#include <boost/algorithm/string/classification.hpp>
+#include <boost/algorithm/string/constants.hpp>
+#include <boost/algorithm/string/split.hpp>
+
+#include <algorithm>
+#include <string>
+
 namespace pandaproxy::rest {
 
+namespace {
+net::unresolved_address address_from_url(std::string_view url) {
+    std::vector<std::string> parts;
+    boost::split(parts, url, boost::is_any_of("/:"), boost::token_compress_on);
+
+    // After split, there should be 4 strings in vector
+    // [http, domain, port, trailing route]. Anything else
+    // is an unacceptable url.
+    if (parts.size() != 4) {
+        throw parse::error(parse::error_code::not_acceptable, "Invalid URL");
+    }
+
+    auto res = parse::from_chars<uint16_t>{}(parts[2]);
+    if (res.has_error()) {
+        throw parse::error(
+          parse::error_code::not_acceptable, "INT conversion fail");
```

Comment:
It may be worth adding some context, we know we're trying to use `part[2]` as a port number, and we know we're decoding a uri.

Reply:
I'm going to look into a 3rd party url decoder because there so many ways for this to fail...

Reply:
In particular `boost::is_any_of("/:")` doesn't work with an ipv6 ip address.

Reply:
Need to change error message